ddev/src/ddev/cli/meta/scripts/scripts/serve.py

```diff
@@ -11,21 +11,30 @@
 else:
     CONTENT_TYPE = 'plain/text; charset=utf-8'
 
-METRICS_FILE = "/tmp/metrics.txt"
+PORT = 8080
 
 
 class OpenMetricsHandler(BaseHTTPRequestHandler):
     def do_GET(self):
+        global current_payload
```

Comment:
Uff, this wasn't easy to read considering the globals are declared a) later and b) conditionally.

I'd suggest defining an `__init__` where you initialize `current_payload` as a class attribute and pass `payloads` as argument.

You can then do: `HTTPServer(("", PORT), functools.partial(OpenMetricsHandler, payloads))` in the main part.

Similar to the cli args parsing though, I'm starting to get a vibe that we're sliding into territory where something like `flask` would be a better fit.

Reply:
I dropped the global variable for the payloads as you suggested. I can't do it for `current_payload` because a new instance of this class is instantiated each time the server is hit.

> Similar to the cli args parsing though, I'm starting to get a vibe that we're sliding into territory where something like flask would be a better fit.

I think we could consider this approach if we keep adding new stuff to this, where the need to have a separate container will arise. I personally think it's "good enough" for now and I don't see any other use cases at the moment